File: Garcia-Bautista-Jonathan_Ulises/Procesos/C++/execl.cpp
#include <iostream>
#include <unistd.h>
#include <sys/types.h>

/*
    g++ -o execl execl.cpp
*/
using namespace std;

class Process {
public:
    void showCurrentPidAndRunPs() {
        pid_t pid = getpid();
        cout << "El pid actual es: " << pid << endl << endl;
        execl("/usr/bin/ps", "ps", "-fea", (char *)NULL);
        cout << "Nunca me verán" << endl;
    }
};

int main() {
    Process process;
    process.showCurrentPidAndRunPs();
    return 0;
}

File: Garcia-Bautista-Jonathan_Ulises/Procesos/Java/Execl.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Execl {
    public static void main(String[] args) {
        // Obtener el PID actual
        String pid = ProcessHandle.current().pid() + "";
        System.out.println("El pid actual es: " + pid);

        try {
            // Ejecutar el comando 'ps -fea'
            Process process = new ProcessBuilder("/usr/bin/ps", "-fea").start();

            // Capturar la salida del proceso
            BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()));
            String line;

            // Leer la salida línea por línea e imprimirla
            while ((line = reader.readLine()) != null) {
                System.out.println(line);
            }

            // Esperar a que el proceso termine y obtener el código de salida
            int exitCode = process.waitFor();

            // Verificar si el comando falló
            if (exitCode != 0) {
                System.out.println("El comando falló con el código de salida: " + exitCode);
                System.out.println("Nunca me verán");
            }

        } catch (IOException | InterruptedException e) {
            // Capturar excepciones y mostrar mensaje
            System.out.println("Error al ejecutar el comando.");
            System.out.println("Nunca me verán");
            e.printStackTrace();
        }
    }
}
